### PR TITLE
Set Random Battle singles levels by species

### PR DESCRIPTION
**Pull request by Sigmatecture**
In preparation for a new level scaling system by species rather than tiers:
global-types.ts: Add randomBattleLevel field to SpeciesFormatsData interface
dex-data.ts: Add randomBattleLevel field to Species class
formats-data.ts: Instantiate this field for each species according to current system
random-teams.ts: Set the Random Battle singles levels by species

### CODE REVIEW

**Comment by DaWoblefet** [comment]
Hey @Sigmatecture can you please not open and close a dozen PRs? Just fix the current one you're working on.

**Comment by Sigmatecture** [comment]
Sorry for the notifications, guys. I'll just keep this one open.

File: data/formats-data.ts [hunk 1]
```diff
@@ -3791,26 +4024,28 @@ export const BattleFormatsData: {[k: string]: SpeciesFormatsData} = {
...
+		randomBattleLevel: 88,
 		randomDoubleBattleMoves: ["destinybond", "disable", "foulplay", "leechseed", "painsplit", "powerwhip", "willowisp"],
 		tier: "(PU)",
 		doublesTier: "(DUU)",
 	},
 	gourgeistsmall: {
 		randomBattleMoves: ["leechseed", "powerwhip", "shadowsneak", "substitute", "willowisp"],
+		randomBattleLevel: 86,
 		randomDoubleBattleMoves: ["destinybond", "disable", "foulplay", "leechseed", "painsplit", "powerwhip", "willowisp"],
-		unreleasedHidden: true,
```
Comment (TheImmortal) [inline data/formats-data.ts:3801]: Re-add this.

File: data/formats-data.ts [hunk 2]
```diff
@@ -3791,26 +4024,28 @@ export const BattleFormatsData: {[k: string]: SpeciesFormatsData} = {
...
 		tier: "NU",
 		doublesTier: "(DUU)",
 	},
 	gourgeistlarge: {
 		randomBattleMoves: ["leechseed", "powerwhip", "shadowsneak", "substitute", "willowisp"],
+		randomBattleLevel: 88,
 		randomDoubleBattleMoves: ["powerwhip", "protect", "shadowclaw", "shadowsneak", "synthesis", "trickroom"],
-		unreleasedHidden: true,
```
Comment (TheImmortal) [inline data/formats-data.ts:3808]: See above.

File: data/formats-data.ts [hunk 3]
```diff
@@ -3829,28 +4065,32 @@ export const BattleFormatsData: {[k: string]: SpeciesFormatsData} = {
...
 	xerneas: {
 		randomBattleMoves: ["calmmind", "focusblast", "moonblast", "psyshock", "thunderbolt"],
+		randomBattleLevel: 72,
 		randomDoubleBattleMoves: ["calmmind", "closecombat", "dazzlinggleam", "healpulse", "moonblast", "protect", "psyshock", "thunderbolt"],
 		isNonstandard: "Past",
 		tier: "Illegal",
 	},
 	xerneasneutral: {
-		isNonstandard: "Custom", // can't be used in battle
+		isNonstandard: "Unobtainable",
```
Comment (TheImmortal) [inline data/formats-data.ts:4081]: Revert this.

File: data/formats-data.ts [hunk 4]
```diff
@@ -4504,8 +4800,8 @@ export const BattleFormatsData: {[k: string]: SpeciesFormatsData} = {
 	},
 	cinderace: {
 		randomBattleMoves: ["courtchange", "gunkshot", "highjumpkick", "pyroball", "uturn", "zenheadbutt"],
+		randomBattleLevel: 80,
 		randomDoubleBattleMoves: ["courtchange", "gunkshot", "highjumpkick", "ironhead", "protect", "pyroball", "uturn", "zenheadbutt"],
-		unreleasedHidden: true,
```
Comment (TheImmortal) [inline data/formats-data.ts:4508]: Hello.

File: data/formats-data.ts [hunk 5]
```diff
@@ -4489,8 +4785,8 @@ export const BattleFormatsData: {[k: string]: SpeciesFormatsData} = {
 	},
 	rillaboom: {
 		randomBattleMoves: ["bulkup", "drumbeating", "highhorsepower", "knockoff", "substitute", "uturn", "woodhammer"],
+		randomBattleLevel: 84,
 		randomDoubleBattleMoves: ["drumbeating", "fakeout", "highhorsepower", "superpower", "swordsdance", "uturn"],
-		unreleasedHidden: true,
```
Comment (TheImmortal) [inline data/formats-data.ts:4493]: Hi.

File: data/formats-data.ts [hunk 6]
```diff
@@ -4530,8 +4827,8 @@ export const BattleFormatsData: {[k: string]: SpeciesFormatsData} = {
 	},
 	greedent: {
 		randomBattleMoves: ["bodyslam", "earthquake", "firefang", "payback", "swordsdance"],
+		randomBattleLevel: 88,
 		randomDoubleBattleMoves: ["bodyslam", "crunch", "earthquake", "gyroball", "protect", "swordsdance"],
-		unreleasedHidden: true,
```
Comment (TheImmortal) [inline data/formats-data.ts:4534]: Stop.

File: data/formats-data.ts [hunk 7]
```diff
@@ -4573,8 +4872,8 @@ export const BattleFormatsData: {[k: string]: SpeciesFormatsData} = {
 	},
 	thievul: {
 		randomBattleMoves: ["darkpulse", "foulplay", "grassknot", "nastyplot", "partingshot", "psychic"],
+		randomBattleLevel: 88,
 		randomDoubleBattleMoves: ["faketears", "foulplay", "partingshot", "snarl", "taunt"],
-		unreleasedHidden: true,
```
Comment (TheImmortal) [inline data/formats-data.ts:4577]: This is not okay.

File: data/formats-data.ts [hunk 8]
```diff
@@ -4593,8 +4893,8 @@ export const BattleFormatsData: {[k: string]: SpeciesFormatsData} = {
 	},
 	dubwool: {
 		randomBattleMoves: ["bodypress", "cottonguard", "rest", "sleeptalk"],
+		randomBattleLevel: 88,
 		randomDoubleBattleMoves: ["doubleedge", "protect", "swordsdance", "thunderwave", "wildcharge", "zenheadbutt"],
-		unreleasedHidden: true,
```
Comment (TheImmortal) [inline data/formats-data.ts:4597]: Enough!

File: data/formats-data.ts [hunk 9]
```diff
@@ -4617,8 +4918,8 @@ export const BattleFormatsData: {[k: string]: SpeciesFormatsData} = {
 	},
 	boltund: {
 		randomBattleMoves: ["bulkup", "crunch", "firefang", "playrough", "psychicfangs", "thunderfang", "voltswitch"],
+		randomBattleLevel: 84,
 		randomDoubleBattleMoves: ["crunch", "firefang", "nuzzle", "playrough", "protect", "psychicfangs", "snarl", "thunderfang"],
-		unreleasedHidden: true,
```
Comment (TheImmortal) [inline data/formats-data.ts:4621]: .

File: data/formats-data.ts [hunk 10]
```diff
@@ -4858,45 +5183,48 @@ export const BattleFormatsData: {[k: string]: SpeciesFormatsData} = {
...
 		randomBattleMoves: ["earthquake", "heatcrash", "heavyslam", "powerwhip", "stoneedge"],
+		randomBattleLevel: 84,
 		tier: "(Uber)",
 		doublesTier: "(DUber)",
 	},
 	dracozolt: {
 		randomBattleMoves: ["aerialace", "boltbeak", "earthquake", "lowkick", "outrage"],
+		randomBattleLevel: 82,
 		randomDoubleBattleMoves: ["aerialace", "boltbeak", "dragonclaw", "highhorsepower", "rockslide"],
-		unreleasedHidden: true,
```
Comment (TheImmortal) [inline data/formats-data.ts:4873]: Another.

File: data/formats-data.ts [hunk 11]
```diff
@@ -4858,45 +5183,48 @@ export const BattleFormatsData: {[k: string]: SpeciesFormatsData} = {
...
 		tier: "UUBL",
 		doublesTier: "(DUU)",
 	},
 	arctozolt: {
 		randomBattleMoves: ["bodyslam", "boltbeak", "freezedry", "iciclecrash", "lowkick"],
+		randomBattleLevel: 88,
 		randomDoubleBattleMoves: ["blizzard", "boltbeak", "iciclecrash", "lowkick", "protect"],
-		unreleasedHidden: true,
```
Comment (TheImmortal) [inline data/formats-data.ts:4880]: One more.

**Pull request merged**